templates/default/oauthConfig.erb

```diff
@@ -9,80 +9,90 @@ oauthConfig:
   grantConfig:
     method: <%= node['cookbook-openshift3']['openshift_master_oauth_grant_method'] %>
   identityProviders:
-  - name: <%= node['cookbook-openshift3']['openshift_master_identity_provider'][node['cookbook-openshift3']['oauth_Identity']]['name']%>
-    challenge: <%= node['cookbook-openshift3']['openshift_master_identity_provider'][node['cookbook-openshift3']['oauth_Identity']]['challenge']%>
-    login: <%= node['cookbook-openshift3']['openshift_master_identity_provider'][node['cookbook-openshift3']['oauth_Identity']]['login']%>
+<% for identityProvider in node['cookbook-openshift3']['oauth_Identities'] %>
```

Comment:
I like that your refactor avoids copying the `node['cookbook-openshift3']['oauth_Identity']` again and again in this template, thanks